client/controllers/events/eventFiltration.coffee

```diff
@@ -60,3 +119,24 @@ Template.eventFiltration.events
     instance.$('.other-properties input:checked').each (i, input) ->
       otherProps[instance.removePropPrefix(input.id)] = true
     instance.properties.set(otherProps)
+
+  'change .locations select': (event, instance) ->
+    instance.countryLevel.set(event.target.value)
+
+  'click .location-list li': (event, instance) ->
+    selectedLocations = instance.selectedLocations
+    if selectedLocations.findOne(id: @_id)
```

Comment:
@jgoley That might be a sign that something is getting added to the collection twice or that the same id is being used by multiple locations. Doing a `if not selectedLocations.find(x)` before all your inserts should prevent any duplicates from being added. If it's something else using the id property is okay.